### PR TITLE
ci: removed run on pull request, refactor tagging

### PR DESCRIPTION
**Pull request by loboda4450**
Updated pipelines that are triggered only manually or on push. Updated tagging to either short commit sha or release/tag version.

Built image:
https://hub.docker.com/layers/voltaserve/api/47e4c20/images/sha256-91e93cd7aa83a2f63cae68a0721b1cad27e112b26f384499de124c6cd9c96cb9?context=explore

Closes: #142

### CODE REVIEW

**Review by dsonck92 — APPROVED** [review]

**Review by dsonck92 — APPROVED** [review]
Nice, I assume you tested the functionality. It looks reasonable, but ya never know with bash magic.

File: .github/workflows/build-api.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
         run: |
           name="voltaserve/api"
           version="${{ github.ref_name }}"
-          version="${version}#v"
+          version="${version#v}"
```
Comment (dsonck92) [inline .github/workflows/build-api.yml:51]: Exactly, I definitely didn't catch that.

**Comment by loboda4450** [comment]
Omg, thanks @dsonck92 for sowing a seed of uncertainty, I double checked and found an error that would append a useless "v" to starting tag and leave all other tags with it :D

**Comment by dsonck92** [comment]
Yeeeah, it wouldn't be the first time I would merge something in that would have had an oversight. I'm always kinda careful around these bash/shell things as it's just such a tricky scripting language.

**Comment by dsonck92** [comment]
@loboda4450, would it also be possible to include the manual running trigger? Or am I missing where to click this button. I want to submit a fix that might shed some insight on why something isn't working for me locally, and having a test container would be nice. Of course local pushing to somewhere would be an option as well, but, yeah, convenience.

**Comment by loboda4450** [comment]
Yup, that's why I am laughing at my job that me and bash is not a good pair, I'd go with python, but not for that small things.

Just to ensure, thats what you meant in the issue, right?
```bash
❯ ./script.sh
name: voltaserve/api
inital version: 1.2.3
parsed tags: voltaserve/api:1.2.3,voltaserve/api:1.2,voltaserve/api:1,voltaserve/api:latest
```

**Comment by dsonck92** [comment]
Yes, this seems like the way to go, then people can pull in `api` `api:1` `api:1.2` and if we're specific `api:1.2.3`

**Comment by loboda4450** [comment]
> @loboda4450, would it also be possible to include the manual running trigger? Or am I missing where to click this button. I want to submit a fix that might shed some insight on why something isn't working for me locally, and having a test container would be nice. Of course local pushing to somewhere would be an option as well, but, yeah, convenience.

Could you be more specific? If you're talking about running the pipeline manually, then you just need to go to actions -> pick the pipeline -> run workflow, then pick the branch and you are good to go. But I believe, that to run this specific pipeline you'd need the workflows inside of your branch. I can add a optional input parameter that will allow running main branch pipelines on another branches, but it will not be a dynamic choice, you'd need to provide the branch name by hand, as github supports only fixed values.

**Comment by dsonck92** [comment]
I guess I'm way too spoiled with CI on gitlab, where you just have an integrated runner dialog on the MR where every manual stage is a cog wheel that can be "started"

**Comment by dsonck92** [comment]
> > @loboda4450, would it also be possible to include the manual running trigger? Or am I missing where to click this button. I want to submit a fix that might shed some insight on why something isn't working for me locally, and having a test container would be nice. Of course local pushing to somewhere would be an option as well, but, yeah, convenience.
> 
> Could you be more specific? If you're talking about running the pipeline manually, then you just need to go to actions -> pick the pipeline -> run workflow, then pick the branch and you are good to go. But I believe, that to run this specific pipeline you'd need the workflows inside of your branch. I can add a optional input parameter that will allow running main branch pipelines on another branches, but it will not be a dynamic choice, you'd need to provide the branch name by hand, as github supports only fixed values.

Ah yes, I see now. I guess with the limitations of github, that will work.

**Comment by loboda4450** [comment]
Would you like this input only in build, or also in linters?

**Comment by dsonck92** [comment]
Well, as long as it's possible to trigger the pipeline in some way, I'm good. Whether this is the pipeline tweaks of the PR, or main, I don't mind. As long as the trigger possibility is there.

Basically, if I'm working in a branch based on `main`, I want to be able to kick off an docker build to a hash as docker image tag, so I can run it in my cluster to validate my changes.

**Comment by loboda4450** [comment]
Tbh I'd stick to workflow definitions from main branch, which means, that you'd need to go to actions -> pick workflow -> run workflow and enter your desired branch to build from. That will override the checkout of `main` branch and build and tag the code from the HEAD of provided branch.

**Pull request merged**